cmd/query/app/query_parser.go

```diff
@@ -46,22 +49,62 @@ const (
 var (
 	errMaxDurationGreaterThanMin = fmt.Errorf("'%s' should be greater than '%s'", maxDurationParam, minDurationParam)
 
-	// ErrServiceParameterRequired occurs when no service name is defined
-	ErrServiceParameterRequired = fmt.Errorf("parameter '%s' is required", serviceParam)
+	// errServiceParameterRequired occurs when no service name is defined.
+	errServiceParameterRequired = fmt.Errorf("parameter '%s' is required", serviceParam)
 )
 
-// queryParser handles the parsing of query parameters for traces
-type queryParser struct {
-	traceQueryLookbackDuration time.Duration
-	timeNow                    func() time.Time
-}
+type (
+	// queryParser handles the parsing of query parameters for traces.
+	queryParser struct {
+		traceQueryLookbackDuration time.Duration
+		timeNow                    func() time.Time
+	}
+
+	traceQueryParameters struct {
+		spanstore.TraceQueryParameters
+		traceIDs []model.TraceID
+	}
+
+	dependenciesQueryParameters struct {
+		endTs    time.Time
+		lookback time.Duration
+	}
+
+	durationParser interface {
+		parseDuration(string) (time.Duration, error)
+	}
 
-type traceQueryParameters struct {
-	spanstore.TraceQueryParameters
-	traceIDs []model.TraceID
+	// durationStringParser parses duration strings like "5ms".
+	durationStringParser struct{}
+
+	// durationUnitsParser parses integer durations represented as units of time such as "1000".
+	durationUnitsParser struct {
+		units time.Duration
+	}
+)
+
+func (dsp durationStringParser) parseDuration(s string) (time.Duration, error) {
+	return time.ParseDuration(s)
+}
+func (dup durationUnitsParser) parseDuration(s string) (time.Duration, error) {
+	i, err := strconv.ParseInt(s, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	return time.Duration(i) * (dup.units), nil
 }
 
-// parse takes a request and constructs a model of parameters
+// parseTraceQueryParams takes a request and constructs a model of parameters.
+//
+// Why start/end parameters are expressed in microseconds:
```

Comment:
@yurishkuro I've added your comments here for posterity. Please advise if they would be better placed else where.